tests/integration/operator/ginkgo/installation_test.go

```diff
@@ -0,0 +1,392 @@
+			Expect(k8sClient.Create(context.Background(), porterCfg)).Should(Succeed())
+			credSet := NewCredSet(credSetName, "insecureValue", secretName)
+
+			//Create the porter credset via an AgentAction
+			agentAction := createCredentialSetAgentAction(installNamespace, credSet)
+			pollAA := func() bool { return agentActionPoll(agentAction) }
+			Eventually(pollAA, time.Second*120, time.Second*3).Should(BeTrue())
+			inst := NewInstallation(installName, installNamespace)
+			Expect(k8sClient.Create(ctx, inst)).Should(Succeed())
+
+			// Wait for the job to be created
+			installations := waitForInstallationStarted(ctx, installNamespace, installName)
+			installation := installations.Items[0]
+
+			// Validate that the job succeeded
+			installation = waitForInstallationFinished(ctx, installation)
+
+			// Validate that the installation status was updated
+			validateInstallStatus(inst, porterv1.PhaseFailed)
+		})
+	})
+})
+
+func validateInstallStatus(inst *porterv1.Installation, expectedPhase porterv1.AgentPhase) {
+	ctx := context.Background()
+	instName := types.NamespacedName{Namespace: inst.Namespace, Name: inst.Name}
+	Expect(k8sClient.Get(ctx, instName, inst)).To(Succeed())
+	Expect(apimeta.IsStatusConditionTrue(inst.Status.Conditions, string(porterv1.ConditionScheduled)))
+	Expect(apimeta.IsStatusConditionTrue(inst.Status.Conditions, string(porterv1.ConditionStarted)))
+	Expect(apimeta.IsStatusConditionTrue(inst.Status.Conditions, string(porterv1.ConditionComplete)))
+	Expect(inst.Status.Phase).To(Equal(expectedPhase))
+}
+
+func agentActionPoll(aa *porterv1.AgentAction) bool {
+	aaName := types.NamespacedName{Name: aa.Name, Namespace: aa.Namespace}
+	Expect(k8sClient.Get(context.Background(), aaName, aa)).To(Succeed())
+	return aa.Status.Phase == porterv1.PhaseSucceeded
+}
+
+func NewInstallation(installationName, installationNamespace string) *porterv1.Installation {
+	return &porterv1.Installation{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "porter.sh/v1",
+			Kind:       "Installation",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      installationName,
+			Namespace: installationNamespace,
+		},
+		Spec: porterv1.InstallationSpec{
+			SchemaVersion: "1.0.0",
+			Name:          installationName,
+			Namespace:     installationNamespace,
+			Bundle: porterv1.OCIReferenceParts{
+				Repository: "ghcr.io/bdegeeter/porter-test-me",
+				Version:    "0.2.0",
+			},
+			Parameters:     runtime.RawExtension{Raw: []byte("{\"delay\": \"1\", \"exitStatus\": \"0\"}")},
+			CredentialSets: []string{"test"},
+		},
+	}
+}
+
+func NewCredSet(name, credName, source string) []byte {
+	return []byte(fmt.Sprintf(`# Use the local kind cluster created by mage EnsureTestCluster
+schemaVersion: 1.0.1
+name: %s
+credentials:
+  - name: %s
+    source:
+      secret: %s
+`, name, credName, source))
+}
+
+func waitForInstallationStarted(ctx context.Context, ns, installationName string) porterv1.InstallationList {
```

Comment:
Long term it would be great to find a home for these test helpers. I expect over time we will build up more test helper methods and structs that can be shared between any projects working in kubernetes. Perhaps in a test package in operator or something like that.

It's fine to have it copied for now, just putting out the idea of how I'd like to see this evolve going forward so it's easier for us to share stuff like this.